sotodlib/flags.py

```diff
@@ -82,115 +85,136 @@ def get_glitch_flags(
+        )
+        glitches = core.AxisManager(
+            aman.dets,
             aman.samps,
         )
-        glitches.wrap("glitch_flags", flag, [(0,"dets"),(1,"samps")])
-        glitches.wrap("glitch_detection", smat, [(0, 'dets'), (1,'samps')])
+        glitches.wrap("glitch_flags", flag, [(0, "dets"), (1, "samps")])
+        glitches.wrap("glitch_detection", smat, [(0, "dets"), (1, "samps")])
         return flag, glitches
 
     return flag
 
-def get_trending_flags(aman, max_trend=5*np.pi, n_pieces=1, signal=None,
-                     merge=True, overwrite=True, name='trends',
-                     full_output=False):
-    """ Flag Detectors with trends larger than max_trend. Note, this is really
+
+def get_trending_flags(
+    aman,
+    max_trend=0.01,
+    n_pieces=1,
+    max_samples=500,
+    signal=None,
+    timestamps=None,
+    merge=True,
+    overwrite=True,
+    name="trends",
+    full_output=False,
+):
+    """Flag Detectors with trends larger than max_trend. Note, this is really
     a max-min calculator.
 
     Args:
         aman (AxisManager): the tod
-        max_trend: maxmimum amount to always the detectors to change by.
-            The default is for use with phase units.
-        n_pieces: number of pieces to cut the timestream in to to look for
-                    trends
+        max_trend: Slope at which detectors are unlocked.
+                   The default is for use with phase units.
+        n_pieces: number of pieces to cut the timestream in to to look for trends.
+        max_samples: Maximum samples to compute the slope with.
         signal: Signal to use to generate flags, default is aman.signal.
+        timestamps: Timestamps to use to generate flags, default is aman.timestamps.
         merge (bool): if true, merges the generated flag into aman
         overwrite (bool): if true, write over flag. if false, don't
         name (string): name of flag to add to aman.flags if merge is True
         full_output(bool): if true, returns calculated slope sizes
-        
+
     Returns:
-        flag: RangesMatrix object of glitches
+        cut: RangesMatrix of trending regions
         trends: if full_output is true, calculated slopes and the
-        sample edges where they were calculated. 
+        sample edges where they were calculated.
     """
     if overwrite and name in aman.flags:
         aman.flags.move(name, None)
 
     if signal is None:
         signal = aman.signal
+    signal = np.atleast_2d(signal)
+    if timestamps is None:
+        timestamps = aman.timestamps
+    assert len(timestamps) == signal.shape[1]
+
+    slopes = np.zeros((len(signal), 0))
+    cut = np.zeros((len(signal), 0), dtype=bool)
+    samp_edges = [0]
+    for t, s in zip(
+        np.array_split(timestamps, n_pieces), np.array_split(signal, n_pieces, 1)
+    ):
+        samps = len(t)
+        # Cheap downsampling
+        if len(t) < max_samples:
```

Comment:
this should be if samps > max_samples. We don't need to down sample if the array is shorter

Reply:
Good catch